### PR TITLE
improve: ascanrules: Address SAST (sonar) findings in CodeInjection Rule

### PR DESCRIPTION
**Pull request by kingthorin**
- Reduce try/catch nesting (SocketException is a child of IOException).
- Remove some seemingly pointless comments.
- Return logical comparisons directly instead of within another conditional.
- Use simpler or more realistic math operations.
- Update unit test to account for newer math/behavior.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by thc202** [comment]
The changelog should be updated.

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CodeInjectionScanRule.java [hunk 1]
```diff
@@ -248,64 +227,48 @@ private boolean testPhpInjection(String paramName) {
      * @see #ASP_PAYLOADS
      */
     private boolean testAspInjection(String paramName) {
-        int bignum1 = 100000 + (int) (RAND.nextFloat() * (999999 - 1000000 + 1));
-        int bignum2 = 100000 + (int) (RAND.nextFloat() * (999999 - 1000000 + 1));
+        int bignum1 = Math.abs(RAND.nextInt(Integer.MAX_VALUE));
+        int bignum2 = Math.abs(RAND.nextInt(Integer.MAX_VALUE));
```
Comment (thc202) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CodeInjectionScanRule.java:231]: `abs` is not required when using limit. Maybe the value should be smaller (more like it was before).

Comment (kingthorin) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CodeInjectionScanRule.java:231]: Set local MAX_VALUE constant 999,999. Updated UnitTest to zero or one thousands 'portion' in the regex (? vs *).

Comment (thc202) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CodeInjectionScanRule.java]: Sounds good.

**Comment by kingthorin** [comment]
Done

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Comment by thc202** [comment]
I forgot about the changelog... it still needs to be updated.

**Comment by kingthorin** [comment]
As a fix?
`Code Injection scan rule, fixed an issue where the ASP injection might not be properly tested.`

Something like that? (Even that's a stretch, I think it was working as it was.)

The CHANGELOG already contains a maint note.

**Comment by thc202** [comment]
A fix or a change, should mention that it's now using random numbers.

**Pull request merged**